drivers/i2c/i2c_ll_stm32.c

```diff
@@ -292,6 +292,11 @@ static void i2c_stm32_irq_config_func_##name(struct device *dev)	\
 #define STM32_I2C_INIT(name)						\
 STM32_I2C_IRQ_HANDLER_DECL(name);					\
 									\
+COND_CODE_1(CONFIG_I2C_STM32_V2,					\
```

Comment:
You can make this much more readable (and consistent with other code) as:
```
#if DT_HAS_COMPAT_STATUS_OKAY(i2c_ll_stm32_v2)
#define DEFINE_TIMINGS(name)						\
	static const uint32_t i2c_timings_##name[] =			\
		DT_PROP_OR(DT_NODELABEL(name), timings, {});
#define USE_TIMINGS(name)						\
	.timings = (const struct i2c_config_timing *) i2c_timings_##name, \
	.n_timings = ARRAY_SIZE(i2c_timings_##name),
#else /* V2 */
#define DEFINE_TIMINGS(name)
#define USE_TIMINGS(name)
#endif /* V2 */

#define STM32_I2C_INIT(name)						\
STM32_I2C_IRQ_HANDLER_DECL(name);					\
									\
DEFINE_TIMINGS(name)							\
									\
static const struct i2c_stm32_config i2c_stm32_cfg_##name = {		\
	.i2c = (I2C_TypeDef *)DT_REG_ADDR(DT_NODELABEL(name)),		\
	.pclken = {							\
		.enr = DT_CLOCKS_CELL(DT_NODELABEL(name), bits),	\
		.bus = DT_CLOCKS_CELL(DT_NODELABEL(name), bus),		\
	},								\
	STM32_I2C_IRQ_HANDLER_FUNCTION(name)				\
	.bitrate = DT_PROP(DT_NODELABEL(name), clock_frequency),	\
	USE_TIMINGS(name)						\
};									\
```

However, whatever preprocessor expression is using to select this must match what's done in `i2c_ll_stm32.h`.

Reply:
Ok for the proposals, thanks for providing the code.
Just one point, from the tests I run:
#if DT_HAS_COMPAT_STATUS_OKAY(i2c_ll_stm32_v2)  => does not work (condition is not met)
while 
#ifdef CONFIG_I2C_STM32_V2   => works OK
@erwango and @pabigot both think that it should work, am I missing something ? 
I'm on master form Aug 19th, is this something that has been introduced in the past few days ?

Reply:
For now, I have pushed the format changes except I use #ifdef CONFIG_I2C_STM32_V2

Reply:
@LMESTM during your trial, are you sure you have a matching dts node enabled ?

Reply:
> @LMESTM during your trial, are you sure you have a matching dts node enabled ?

Well the timings handling is only done in I2C v2 driver. 
As said, timing is applied / not applied depending on the change below only: 
#if DT_HAS_COMPAT_STATUS_OKAY(i2c_ll_stm32_v2)
#ifdef CONFIG_I2C_STM32_V2 => works OK

Am I expect the driver type to be defined by default, do I need to add something in DT to get i2c_ll_stm32_v2 defined ?

Reply:
ok I found out the proper form: 
#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_i2c_v2)
and I pushed a new update accordingly